### PR TITLE
actual query cost and requested query cost are now returned with the …

### PR DESCRIPTION
**Pull request by ozzyonfire**
…callGraphqlLimits event

### CODE REVIEW

**Comment by ozzyonfire** [comment]
Sorry, I haven't yet tested this on a production shopify store. I just wanted to push the branch into my forked repo. I will do a live test now.

**Comment by ozzyonfire** [comment]
Fixes #489 #490. Related to #451

**Comment by ozzyonfire** [comment]
Tested with live production store and everything looks good.

File: index.js [hunk 1]
```diff
@@ -193,17 +193,24 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
...
  * @param {String} throttle The status returned in the GraphQL response
+ * @param {Number} requestedQueryCost The requested cost from the GraphQL query
+ * @param {Number} actualQueryCost The actual cost from the GraphQL query
  * @private
  */
-Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits(throttle) {
+Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits(
+  throttle,
+  requestedQueryCost,
+  actualQueryCost
```
Comment (lpinca) [inline index.js:203]: I would pass only `extensions.cost` here, it includes everything. Also please add `actualQueryCost` and `requestedQueryCost` in the object on line 58.

**Comment by lpinca** [comment]
The TypeScript type definitions should also be updated.

File: index.js [hunk 2]
```diff
@@ -192,18 +194,24 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
 /**
  * Updates GraphQL API call limits.
  *
- * @param {String} throttle The status returned in the GraphQL response
+ * @param {String} cost The status returned in the GraphQL response
```
Comment (lpinca) [inline index.js:197]:
```suggestion
 * @param {Object} cost The cost object returned in the GraphQL response
```

File: index.js [hunk 3]
```diff
@@ -192,18 +194,24 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
...
  * @private
  */
-Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits(throttle) {
+Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits(cost) {
```
Comment (lpinca) [inline index.js:200]:
```suggestion
Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits({ throttleStatus, actualQueryCost, requestedQueryCost }) {
```

File: index.js [hunk 4]
```diff
@@ -192,18 +194,24 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
...
+  const {
+    throttleStatus: throttle,
+    actualQueryCost,
+    requestedQueryCost
+  } = cost;
```
Comment (lpinca) [inline index.js:205]:
```suggestion
```

File: index.js [hunk 5]
```diff
@@ -192,18 +194,24 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
...
   if (!throttle) return;
 
   const limits = this.callGraphqlLimits;
-
   limits.current = throttle.maximumAvailable - throttle.currentlyAvailable;
   limits.remaining = throttle.currentlyAvailable;
   limits.restoreRate = throttle.restoreRate;
   limits.max = throttle.maximumAvailable;
```
Comment (lpinca) [inline index.js:212]:
```suggestion
  limits.current = throttleStatus.maximumAvailable - throttleStatus.currentlyAvailable;
  limits.remaining = throttleStatus.currentlyAvailable;
  limits.restoreRate = throttleStatus.restoreRate;
  limits.max = throttleStatus.maximumAvailable;
```

File: index.js [hunk 6]
```diff
@@ -192,18 +194,19 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
 /**
  * Updates GraphQL API call limits.
  *
- * @param {String} throttle The status returned in the GraphQL response
+ * @param {Object} cost The cost object returned in the GraphQL response
  * @private
  */
-Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits(throttle) {
+Shopify.prototype.updateGraphqlLimits = function updateGraphqlLimits({ throttleStatus, actualQueryCost, requestedQueryCost }) {
   if (!throttle) return;
```
Comment (lpinca) [inline index.js:201]:
```suggestion
  if (!throttleStatus) return;
```

**Review by lpinca — APPROVED** [review]

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**